FreeRTOS_UDP_IP.c

```diff
@@ -573,6 +583,27 @@ static BaseType_t prvHandleUDPPacketWithoutSocket( NetworkBufferDescriptor_t * p
         else
     #endif /* ipconfigUSE_LLMNR */
 
+    #if ( ipconfigUSE_MDNS == 1 )
+        /* A MDNS request, check for the destination port. */
+        if( ( usPort == FreeRTOS_ntohs( ipMDNS_PORT ) ) ||
+            ( pxProtocolHeaders->xUDPHeader.usSourcePort == FreeRTOS_ntohs( ipMDNS_PORT ) ) )
+        {
+            #if ( ipconfigUSE_IPv6 != 0 )
+                if( pxUDPPacket->xEthernetHeader.usFrameType == ipIPv6_FRAME_TYPE )
+                {
```

Comment:
There is a race condition between our PR's: your #338 is faster than my #337 !
Are you asking if we must call `vARPRefreshCacheEntry()`? I think yes, because this is a packet for this device.
We should only check if the target address is not a multicast address.

Reply:
Then we should check that and update the cache IMO. What do you think?

Reply:
Can you look at the above comment?

Reply:
This PR is only about DNS and local mDNS.
When this is merged, I would like to port your latest IPv4 PR's to IPv6/multi. And then I will also look at updating the ND, agreed?